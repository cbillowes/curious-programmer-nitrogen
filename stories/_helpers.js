import React from 'react'
import Layout from '../src/components/layout'
import { SmallSpacer } from '../src/components/spacer'

const Page = ({ className, children }) => {
  return (
    <Layout
      className={className}
    >
      <SmallSpacer />
      {children}
    </Layout>
  )
}

export default Page