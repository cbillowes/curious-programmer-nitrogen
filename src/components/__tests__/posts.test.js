import React from 'react'
import Posts from '../posts'
import { getSnapshot } from './_helpers'

describe(`Posts`, () => {
  const edges = [
    {
      node: {
        excerpt: ``,
        timeToRead: `12`,
        html: `,`,
        fields: {
          slug: `/dont-panic`,
          date: `2019-02-01T00:00:00.000Z`,
          number: 42,
        },
        frontmatter: {
          title: ``,
          author: `Clarice Bouwer`,
          blurb: ``,
          ttr: `42`,
          tags: [`don't`, `panic`],
        },
      },
    },
    {
      node: {
        excerpt: ``,
        timeToRead: `12`,
        html: `,`,
        fields: {
          slug: `/bring-your-own-towel`,
          date: `2019-02-01T00:00:00.000Z`,
          number: 42,
        },
        frontmatter: {
          title: ``,
          author: `Clarice Bouwer`,
          blurb: ``,
          ttr: `42`,
          tags: [`towel`, `panic`],
        },
      },
    },
  ]

  it(`should render with default word limit`, () => {
    const posts = (
      <Posts
        edges={edges}
      />
    )
    const tree = getSnapshot(posts)
    expect(tree).toMatchSnapshot()
  })

  it(`should render with specified word limit of 5`, () => {
    const posts = (
      <Posts
        edges={edges}
        limit={1}
      />
    )
    const tree = getSnapshot(posts)
    expect(tree).toMatchSnapshot()
  })
})
