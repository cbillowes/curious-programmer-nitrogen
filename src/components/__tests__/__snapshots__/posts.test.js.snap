// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts should render with default word limit 1`] = `
<div
  className="posts-container"
>
  <div
    className="actions"
  >
    <button
      className="button multi-column active"
      onClick={[Function]}
    >
      <img
        alt="Display in a verbose grid."
        src="/grid.svg"
      />
    </button>
    <button
      className="button single-column active"
      onClick={[Function]}
    >
      <img
        alt="Display in a verbose list."
        src="/vertical-list.svg"
      />
    </button>
    <button
      className="button "
      onClick={[Function]}
    >
      <img
        alt="Display in a compact list."
        src="/listing.svg"
      />
    </button>
    <span
      className="search "
    >
      <button
        className="button search-btn"
      >
        <img
          alt="Search"
          src="/search.svg"
        />
      </button>
      <button
        className="button reset-btn"
        onClick={[Function]}
      >
        <img
          alt="Start over"
          src="/close.svg"
        />
      </button>
      <input
        className="search-bar"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        placeholder="Search a title"
        type="text"
        value=""
      />
      <div
        className="search-results"
      >
        <span
          className="result"
        >
          2
           results.
        </span>
      </div>
    </span>
  </div>
  <div
    className="posts grid"
  >
    <section
      className="post preview"
    >
      <div
        className="post number ribbon"
      >
        <div
          className="backdrop"
        >
          <div
            className="text"
          >
            #
            42
          </div>
        </div>
      </div>
      <div
        className="thumbnail"
      >
         
        <a
          className="slug"
          href="/dont-panic"
        >
          <div
            className=" gatsby-image-wrapper"
            style={
              Object {
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            <div
              aria-hidden={true}
              style={
                Object {
                  "paddingBottom": "NaN%",
                  "width": "100%",
                }
              }
            />
            <picture>
              <source />
              <img
                alt=""
                loading="lazy"
                onLoad={[Function]}
                style={
                  Object {
                    "height": "100%",
                    "left": 0,
                    "objectFit": "cover",
                    "objectPosition": "center",
                    "opacity": 0,
                    "position": "absolute",
                    "top": 0,
                    "transition": "opacity 500ms",
                    "width": "100%",
                  }
                }
              />
            </picture>
            <noscript
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<picture><source srcset=\\"undefined\\" /><img loading=\\"lazy\\" src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
                }
              }
            />
          </div>
        </a>
         
         
        <a
          className="badge linked"
          title="Download free do whatever you want high-resolution photos from Dlanor S"
        >
          <span
            className="source-logo"
          >
            <img
              alt="unsplash"
              src="/unsplash.svg"
            />
          </span>
          <span
            className="attribute"
          >
            Dlanor S
          </span>
        </a>
         
      </div>
      <h2>
         
        <a
          href="/dont-panic"
          title=""
        >
          
        </a>
         
      </h2>
      <div
        className="blurb"
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": ",",
            }
          }
        />
      </div>
      <div
        className="metadata"
      >
        <div
          className="metadata"
        >
          posted on
           
          <time
            dateTime={"2019-02-01T00:00:00.000Z"}
          >
            Friday, 1 February 2019
          </time>
           
          <span>
            by Clarice Bouwer
          </span>
           
          <span>
            (Est. 42 minute read)
          </span>
        </div>
      </div>
      <div
        className="tags"
      >
         
        <a
          className="tag"
          href="/tag/dont/"
          title="don't"
        >
          don't
        </a>
         
         
        <a
          className="tag"
          href="/tag/panic/"
          title="panic"
        >
          panic
        </a>
         
      </div>
    </section>
    <section
      className="post preview"
    >
      <div
        className="post number ribbon"
      >
        <div
          className="backdrop"
        >
          <div
            className="text"
          >
            #
            42
          </div>
        </div>
      </div>
      <div
        className="thumbnail"
      >
         
        <a
          className="slug"
          href="/bring-your-own-towel"
        >
          <div
            className=" gatsby-image-wrapper"
            style={
              Object {
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            <div
              aria-hidden={true}
              style={
                Object {
                  "paddingBottom": "NaN%",
                  "width": "100%",
                }
              }
            />
            <picture>
              <source />
              <img
                alt=""
                loading="lazy"
                onLoad={[Function]}
                style={
                  Object {
                    "height": "100%",
                    "left": 0,
                    "objectFit": "cover",
                    "objectPosition": "center",
                    "opacity": 0,
                    "position": "absolute",
                    "top": 0,
                    "transition": "opacity 500ms",
                    "width": "100%",
                  }
                }
              />
            </picture>
            <noscript
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<picture><source srcset=\\"undefined\\" /><img loading=\\"lazy\\" src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
                }
              }
            />
          </div>
        </a>
         
         
        <a
          className="badge linked"
          title="Download free do whatever you want high-resolution photos from Dlanor S"
        >
          <span
            className="source-logo"
          >
            <img
              alt="unsplash"
              src="/unsplash.svg"
            />
          </span>
          <span
            className="attribute"
          >
            Dlanor S
          </span>
        </a>
         
      </div>
      <h2>
         
        <a
          href="/bring-your-own-towel"
          title=""
        >
          
        </a>
         
      </h2>
      <div
        className="blurb"
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": ",",
            }
          }
        />
      </div>
      <div
        className="metadata"
      >
        <div
          className="metadata"
        >
          posted on
           
          <time
            dateTime={"2019-02-01T00:00:00.000Z"}
          >
            Friday, 1 February 2019
          </time>
           
          <span>
            by Clarice Bouwer
          </span>
           
          <span>
            (Est. 42 minute read)
          </span>
        </div>
      </div>
      <div
        className="tags"
      >
         
        <a
          className="tag"
          href="/tag/panic/"
          title="panic"
        >
          panic
        </a>
         
         
        <a
          className="tag"
          href="/tag/towel/"
          title="towel"
        >
          towel
        </a>
         
      </div>
    </section>
  </div>
</div>
`;

exports[`Posts should render with specified word limit of 5 1`] = `
<div
  className="posts-container"
>
  <div
    className="actions"
  >
    <button
      className="button multi-column active"
      onClick={[Function]}
    >
      <img
        alt="Display in a verbose grid."
        src="/grid.svg"
      />
    </button>
    <button
      className="button single-column active"
      onClick={[Function]}
    >
      <img
        alt="Display in a verbose list."
        src="/vertical-list.svg"
      />
    </button>
    <button
      className="button "
      onClick={[Function]}
    >
      <img
        alt="Display in a compact list."
        src="/listing.svg"
      />
    </button>
    <span
      className="search "
    >
      <button
        className="button search-btn"
      >
        <img
          alt="Search"
          src="/search.svg"
        />
      </button>
      <button
        className="button reset-btn"
        onClick={[Function]}
      >
        <img
          alt="Start over"
          src="/close.svg"
        />
      </button>
      <input
        className="search-bar"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        placeholder="Search a title"
        type="text"
        value=""
      />
      <div
        className="search-results"
      >
        <span
          className="result"
        >
          2
           results.
        </span>
      </div>
    </span>
  </div>
  <div
    className="posts grid"
  >
    <section
      className="post preview"
    >
      <div
        className="post number ribbon"
      >
        <div
          className="backdrop"
        >
          <div
            className="text"
          >
            #
            42
          </div>
        </div>
      </div>
      <div
        className="thumbnail"
      >
         
        <a
          className="slug"
          href="/dont-panic"
        >
          <div
            className=" gatsby-image-wrapper"
            style={
              Object {
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            <div
              aria-hidden={true}
              style={
                Object {
                  "paddingBottom": "NaN%",
                  "width": "100%",
                }
              }
            />
            <picture>
              <source />
              <img
                alt=""
                loading="lazy"
                onLoad={[Function]}
                style={
                  Object {
                    "height": "100%",
                    "left": 0,
                    "objectFit": "cover",
                    "objectPosition": "center",
                    "opacity": 0,
                    "position": "absolute",
                    "top": 0,
                    "transition": "opacity 500ms",
                    "width": "100%",
                  }
                }
              />
            </picture>
            <noscript
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<picture><source srcset=\\"undefined\\" /><img loading=\\"lazy\\" src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
                }
              }
            />
          </div>
        </a>
         
         
        <a
          className="badge linked"
          title="Download free do whatever you want high-resolution photos from Dlanor S"
        >
          <span
            className="source-logo"
          >
            <img
              alt="unsplash"
              src="/unsplash.svg"
            />
          </span>
          <span
            className="attribute"
          >
            Dlanor S
          </span>
        </a>
         
      </div>
      <h2>
         
        <a
          href="/dont-panic"
          title=""
        >
          
        </a>
         
      </h2>
      <div
        className="blurb"
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": ",",
            }
          }
        />
      </div>
      <div
        className="metadata"
      >
        <div
          className="metadata"
        >
          posted on
           
          <time
            dateTime={"2019-02-01T00:00:00.000Z"}
          >
            Friday, 1 February 2019
          </time>
           
          <span>
            by Clarice Bouwer
          </span>
           
          <span>
            (Est. 42 minute read)
          </span>
        </div>
      </div>
      <div
        className="tags"
      >
         
        <a
          className="tag"
          href="/tag/dont/"
          title="don't"
        >
          don't
        </a>
         
         
        <a
          className="tag"
          href="/tag/panic/"
          title="panic"
        >
          panic
        </a>
         
      </div>
    </section>
    <section
      className="post preview"
    >
      <div
        className="post number ribbon"
      >
        <div
          className="backdrop"
        >
          <div
            className="text"
          >
            #
            42
          </div>
        </div>
      </div>
      <div
        className="thumbnail"
      >
         
        <a
          className="slug"
          href="/bring-your-own-towel"
        >
          <div
            className=" gatsby-image-wrapper"
            style={
              Object {
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            <div
              aria-hidden={true}
              style={
                Object {
                  "paddingBottom": "NaN%",
                  "width": "100%",
                }
              }
            />
            <picture>
              <source />
              <img
                alt=""
                loading="lazy"
                onLoad={[Function]}
                style={
                  Object {
                    "height": "100%",
                    "left": 0,
                    "objectFit": "cover",
                    "objectPosition": "center",
                    "opacity": 0,
                    "position": "absolute",
                    "top": 0,
                    "transition": "opacity 500ms",
                    "width": "100%",
                  }
                }
              />
            </picture>
            <noscript
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<picture><source srcset=\\"undefined\\" /><img loading=\\"lazy\\" src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
                }
              }
            />
          </div>
        </a>
         
         
        <a
          className="badge linked"
          title="Download free do whatever you want high-resolution photos from Dlanor S"
        >
          <span
            className="source-logo"
          >
            <img
              alt="unsplash"
              src="/unsplash.svg"
            />
          </span>
          <span
            className="attribute"
          >
            Dlanor S
          </span>
        </a>
         
      </div>
      <h2>
         
        <a
          href="/bring-your-own-towel"
          title=""
        >
          
        </a>
         
      </h2>
      <div
        className="blurb"
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": ",",
            }
          }
        />
      </div>
      <div
        className="metadata"
      >
        <div
          className="metadata"
        >
          posted on
           
          <time
            dateTime={"2019-02-01T00:00:00.000Z"}
          >
            Friday, 1 February 2019
          </time>
           
          <span>
            by Clarice Bouwer
          </span>
           
          <span>
            (Est. 42 minute read)
          </span>
        </div>
      </div>
      <div
        className="tags"
      >
         
        <a
          className="tag"
          href="/tag/panic/"
          title="panic"
        >
          panic
        </a>
         
         
        <a
          className="tag"
          href="/tag/towel/"
          title="towel"
        >
          towel
        </a>
         
      </div>
    </section>
  </div>
</div>
`;
